src/wallet/wallet.h

```diff
@@ -926,9 +926,9 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
 
     // Returns a list of unspent and verified coins, I.E. coins which are ready
     // to be spent.
-    std::list<CSigmaEntry> GetAvailableCoins(const CCoinControl *coinControl = NULL, bool includeUnsafe = false) const;
+    std::list<CSigmaEntry> GetAvailableCoins(const CCoinControl *coinControl = NULL, bool includeUnsafe = false, bool forEstimation = false) const;
 
-    std::list<CLelantusEntry> GetAvailableLelantusCoins(const CCoinControl *coinControl = NULL, bool includeUnsafe = false) const;
+    std::list<CLelantusEntry> GetAvailableLelantusCoins(const CCoinControl *coinControl = NULL, bool includeUnsafe = false, bool forEstimation = false) const;
```

Comment:
Please consider adding another function for getting an estimates (`GetAvailableLelantusCoinValues` ?). It gets all the coins with their values (no need in secret data).
It would be much better than propagating `forEstimate` flag down the API chain

Reply:
I did not go with that way as much part of the code will be copied.